build/azure-pipelines/cli/cli-compile-and-publish.yml

```diff
@@ -106,3 +106,25 @@ steps:
         - publish: $(Build.ArtifactStagingDirectory)/${{ parameters.VSCODE_CLI_ARTIFACT }}.tar.gz
           artifact: ${{ parameters.VSCODE_CLI_ARTIFACT }}
           displayName: Publish ${{ parameters.VSCODE_CLI_ARTIFACT }} artifact
+
+    # Make a folder for the SBOM for the specific artifact
+    - ${{ if contains(parameters.VSCODE_CLI_TARGET, '-windows-') }}:
+      - powershell: mkdir $(Build.ArtifactStagingDirectory)/sbom_${{ parameters.VSCODE_CLI_ARTIFACT }}
+        displayName: Make folder for SBOM (Windows)
+
+    - ${{ else }}:
+      - script: mkdir $(Build.ArtifactStagingDirectory)/sbom_${{ parameters.VSCODE_CLI_ARTIFACT }}
+        displayName: Make folder for SBOM (non-Windows)
+
+    # The if cases above are for different OSes,
+    # but we're still in the branch where the cli is being published in general.
+    # Generate and publish an SBOM.
+    - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
```

Comment:
Does this make an SBOM for the entire client? Should this stage be scoped to only make an SBOM for the ./cli folder?

Reply:
I noticed that the win32 client and server SBOMs both span the entire workspace. I can restrict the SBOM to only be generated for the CLI folder, but I wonder whether the server SBOM should also be restricted in scope?

Reply:
That would make sense to me. I don't think it necessarily _hurts_, it just means any governance alert will ping additional targets as false-positives, but being more tailored would be nice

Reply:
Just checked the generated SBOM and you're right that the SBOM should be scoped to only the cli folder. I'll also see if I can restrict the server SBOM.